r-build.md

```diff
@@ -0,0 +1,177 @@
+ 5. [Create a portable copy of your R installation that can be brought along with your jobs](#5-create-a-portable-copy-of-your-r-installation)
+ 6. [Use your custom R installation in your jobs](#6-use-your-custom-r-installation-in-your-jobs) 
+
+# Compile R
+
+## 1. Download R Source Code
+
+Before running any commands on CHTC, use a browser to get the source code 
+for your desired version of R from [CRAN](https://cran.r-project.org) 
+Under "Source Code for all Platforms", find the R-#.#.#.tar.gz file for your 
+desired version of R and download it to your computer before transferring a copy 
+to your `/home` directory on the submit server. 
+
+## 2. Submit An Interactive Job To A Build Server
+
+Creating an R installation can be computationally intensive and should not be 
+performed on the submit server. Instead, you will create your installation 
+on a CHTC build server by using an interactive job. 
+
+The interactive job is essentially a job without an executable; 
+you will be the one running the commands, instead (in this case, to install R).
+Like a regular HTCondor job, once you finish your R installation on the build server, 
+the output files (your completed portable R installation) will be transferred back to 
+the submit server (so that you can use the R installation for later jobs). 
+
+To submit an interactive job, create a submit file called `build.sub` as shown below:
+
+``` {.sub}
+# build.sub
+# Software build file
+
+universe = vanilla
+log = interactive.log
+
+# change the name of the file to be the name of your source code
+transfer_input_files = R-#.#.#.tar.gz
+
++IsBuildJob = true
+requirements = (OpSysMajorVer =?= 7)
+request_cpus = 1
+request_memory = 4GB
+request_disk = 4GB
+
+queue
+```
+
+Be sure to modify the above `transfer_input_files` statement with the appropriate 
+name of the R source code that you downloaded in the previous ste.
+
+Once this submit file is created, you will start the interactive job by
+running the following command:
+
+``` 
+[alice@submit]$ condor_submit -i build.sub
+```
+{:.term}
+
+The interactive build job should start in about a minute. Once it has
+started, the job has a time limit of four hours.
+
+## 3. Compile Base R From Source Code
+
+Once the interactive job starts, R can be compiled. The general workflow will be 
+to first run the R configuration script followed by `make` and `make install`. 
+
+To install R, first create a directory that will hold your R installation (e.g. `R/`) 
+, then extract all of the source code files from R-#.#.#.tar.gz and move into the source code directory:
+
+``` 
+[alice@build]$ mkdir R/
+[alice@build]$ tar xzf R-#.#.#.tar.gz
+[alice@build]$ cd R-#.#.#/
+```
+{:.term}
+
+From the source code directory, run the following commands. **The `make` step may take a while to 
+complete!**: 
+
+```
+[alice@build]$ ./configure --prefix=$_CONDOR_SCRATCH_DIR/R
+[alice@build]$ make
+[alice@build]$ make install
```

Comment:
Maybe `make && make install` to reduce the number of commands? But also noodlier. No preference, just an option.

Reply:
Also got an error on this step:
```
installing doc ...
/usr/bin/install: cannot stat ‘NEWS.2.rds’: No such file or directory
/usr/bin/install: cannot stat ‘NEWS.3.rds’: No such file or directory
make[1]: *** [install-sources2] Error 1
make[1]: Leaving directory `/var/lib/condor/execute/slot1/dir_18156/R-4.0.1/doc'
make: *** [install] Error 1
```

I did a `make clean` and tried again, and got
```
installing doc ...
/usr/bin/install: cannot stat ‘NEWS.pdf’: No such file or directory
/usr/bin/install: cannot stat ‘NEWS.pdf’: No such file or directory
/usr/bin/install: cannot stat ‘NEWS.2.rds’: No such file or directory
/usr/bin/install: cannot stat ‘NEWS.3.rds’: No such file or directory
make[1]: *** [install-sources2] Error 1
make[1]: Leaving directory `/var/lib/condor/execute/slot1/dir_18156/R-4.0.1/doc'
make: *** [install] Error 1
```
so... not sure what's up with that.

Reply:
yep - Eric ran into this as well, and ended up just "touching" the files and then installation completed. 

The underlying issue is a latex dependency (to build those files).

Reply:
I personally ran `make; make install` I don't think there's any difference.

Reply:
`make; make install` is only equivalent if `make` succeeds ;)